Code/Framework/AzFramework/Platform/Windows/AzFramework/Windowing/NativeWindow_Windows.cpp

```diff
@@ -192,6 +197,58 @@ namespace AzFramework
         SetWindowTextW(m_win32Handle, titleW.c_str());
     }
 
+    RECT NativeWindowImpl_Win32::GetMonitorRect() const
+    {
+        // Get the monitor on which the window is currently displayed.
+        HMONITOR monitor = MonitorFromWindow(m_win32Handle, MONITOR_DEFAULTTONEAREST);
+        if (!monitor)
+        {
+            AZ_Warning("NativeWindowImpl_Win32::GetMonitorRect", false, "Could not find any monitor.");
+            return { 0, 0, 0, 0 };
+        }
+
+        // Get the dimensions of the display device on which the window is currently displayed.
+        MONITORINFO monitorInfo;
+        memset(&monitorInfo, 0, sizeof(MONITORINFO)); // C4701 potentially uninitialized local variable 'monitorInfo' used
+        monitorInfo.cbSize = sizeof(MONITORINFO);
+        const BOOL success = monitor ? GetMonitorInfo(monitor, &monitorInfo) : FALSE;
+        if (!success)
```

Comment:
I think you already early out if monitor is nullptr on 204
```suggestion
        if (!GetMonitorInfo(monitor, &monitorInfo))
```

Reply:
Great suggestion, though if I don't have any other changes I'll probably leave this one as-is since I've already passed AR.